src/test/setupE2EProcaptcha.ts

```diff
@@ -1,136 +1,252 @@
-        return new ProsopoCaptchaContract(api, abi, address,'procaptcha', 0, pair)
+    set transactionQueue(txQueue: TransactionQueue) {
+        this._transactionQueue = txQueue
     }
-    throw new Error(`Failed to deploy Procaptcha contract: ${JSON.stringify(deployResult)}`)
 
-}
+    async isReady() {
+        try {
+            await this.api.isReady;
+            const alicePair = await getPairAsync(undefined, '//Alice', undefined, 'sr25519', 42)
+            const contract = await this.deployProcaptchaContract(alicePair);
+            console.log("Captcha contract address", contract.address.toString())
+            this.transactionQueue = new TransactionQueue(this.api, alicePair)
 
-export async function procaptchaProviderRegister(contract: ProsopoCaptchaContract, port: number): Promise<void> {
-    try {
-        console.log("Registering Procaptcha provider")
-        const providerRegisterArgs: Parameters<typeof contract.query.providerRegister> = [
-            Array.from(stringToU8a(`http://host.docker.internal:${port}`)),
-            0,
-            Payee.dapp,
-            {
-                value: 1000000000, // minimum value for a captcha provider to be active in the contract
-            },
-        ]
-        await wrapQuery(contract.query.providerRegister, contract.query)(...providerRegisterArgs)
-        await contract.tx.providerRegister(...providerRegisterArgs)
-    } catch(e) {
-        console.error(e)
-        throw new Error (`Failed to register Procaptcha provider: ${JSON.stringify(e, null ,4)}`)
-    }
-}
+            // Set the calling pair to Bob so that he is registered as the provider (the calling account is registered in the contract as the provider)
+            contract.pair = await getPairAsync(undefined, '//Bob', undefined, 'sr25519', 42);
 
-export async function procaptchaProviderSetDataset(contract: ProsopoCaptchaContract): Promise<{ datasetId:Hash, datasetContentId:Hash }> {
-    try {
-        console.log("Setting Procaptcha provider dataset")
-        const dataset = {
-            datasetId: "0x28c1ba9d21c00f2e29c9ace8c46fd7dbfbb6f5a5f516771278635ac3ab88c267", // hashed value of "TESTDATASET"
-            datasetContentId: "0x7d23f5c5e496dc1c9bcf66c62e2ba7a60152f1486ef6032b56809badf0a48427", // hashed value of "TESTDATASETCONTENT"
+            await this.procaptchaProviderRegister(contract, this.port);
+            console.log(`Registered Bob ${contract.pair.address} as provider`)
+            await this.procaptchaProviderSetDataset(contract);
+
+            // any account can register an app site key so using Bob will be fine
+            await this.procaptchaAppRegister(contract, this.siteKey);
+
+            return {contract, contractAddress: contract.address, testAccount: alicePair.address, siteKey: this.siteKey}
+        } catch (e) {
+            throw new Error(`Failed to setup Procaptcha: ${JSON.stringify(e, null, 4)}`)
         }
+    }
```

Comment:
Could you also remove wrapping everything in try-catches from this file as well? 
Callstacks do a better job of showing where the error happened rather than wrapping everything in try-catch, just to throw the error right in the catch block

Reply:
Done.